.eslintrc.js

```diff
@@ -400,10 +400,220 @@ module.exports = {
      * SIEM overrides
      */
     {
-      files: ['x-pack/plugins/siem/**/*.ts'],
+      // front end typescript and javascript files only
+      files: ['x-pack/plugins/siem/public/**/*.{js,ts,tsx}'],
       rules: {
+        'no-restricted-imports': [
+          'error',
+          {
+            // prevents UI code from importing server side code and then webpack including it when doing builds
+            patterns: ['**/server/*'],
+            // prevents UI code from importing NodeJS code
+            paths: [
```

Comment:
Thank-you thank-you -- these are great additions and will save us from future compatibility headaches. For the preventing the client-side NodeJS imports though, could you use [import/no-nodejs-modules](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-nodejs-modules.md) instead of explicitly listing all these?

Reply:
Roger that.